proxy-verifier/proxy-verifier-logic/src/handlers/mod.rs

```diff
@@ -0,0 +1,226 @@
+pub mod solidity_verifier_multi_part;
+pub mod solidity_verifier_standard_json;
+
+pub mod vyper_verifier_multi_part;
+pub mod vyper_verifier_standard_json;
+
+/************************************************/
+
+use crate::{
+    address_details, address_details::AddressDetails, to_hex::ToHex, Error, VerificationResponse,
+};
+use eth_bytecode_db_proto::blockscout::eth_bytecode_db::v2 as eth_bytecode_db_v2;
+use std::future::Future;
+
+async fn process_verification_request<'a, Request, RequestBuilder, Verify, VerifyOutput>(
+    eth_bytecode_db_client: &'a eth_bytecode_db_proto::http_client::Client,
+    contracts: Vec<(&blockscout_client::Client, ethers_core::types::Address)>,
+    request_builder: RequestBuilder,
+    verify: Verify,
+) -> VerificationResponse
+where
+    RequestBuilder: Fn(
+            ethers_core::types::Bytes,
+            eth_bytecode_db_v2::BytecodeType,
+            eth_bytecode_db_v2::VerificationMetadata,
+        ) -> Request
+        + Clone,
+    Verify: Fn(&'a eth_bytecode_db_proto::http_client::Client, Request) -> VerifyOutput + Clone,
+    VerifyOutput: Future<
+        Output = eth_bytecode_db_proto::http_client::Result<eth_bytecode_db_v2::VerifyResponse>,
+    >,
+{
+    let contract_details = retrieve_contract_details(&contracts).await;
+    if let Some(response) = check_invalid_contracts(&contract_details) {
+        return response;
+    }
+
+    let mut results = vec![];
+    for (contract_details, (blockscout_client, contract_address)) in
+        contract_details.into_iter().zip(contracts)
+    {
+        let result = verify_contract(
+            eth_bytecode_db_client,
+            contract_details,
+            request_builder.clone(),
+            verify.clone(),
+        )
+        .await;
+        match result {
+            Ok(()) => {
+                let search_result = search_contract(blockscout_client, contract_address).await;
+                results.push(search_result)
+            }
+            Err(err) if err.is_compilation_failed_error() => {
+                return VerificationResponse::CompilationFailed(err)
+            }
+            Err(err) => results.push(Err(err)),
+        }
+    }
+
+    VerificationResponse::Results(results)
+}
+
+async fn retrieve_contract_details(
+    contracts: &[(&blockscout_client::Client, ethers_core::types::Address)],
+) -> Vec<Result<AddressDetails, Error>> {
+    let mut contract_details = vec![];
+    for (client, contract_address) in contracts {
+        let details = address_details::retrieve_address_details(client, *contract_address).await;
+
+        contract_details.push(details);
+    }
```

Comment:
we did it sequentially because it's not very long requests or we forgot to do it in parallel?

Reply:
No, it may be quite long. I've decided to leave it sequential just for the current initial PR to make the process more straightforward and easier to debug in case of errors

Reply:
I'll add concurrent processing later in one of the next PRs